plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.processor;
+
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.dataset.remote.RemoteInferenceInputDataSet;
+import org.opensearch.ml.common.input.MLInput;
+import org.opensearch.ml.common.output.model.ModelTensorOutput;
+import org.opensearch.ml.common.transport.prediction.MLPredictionTaskRequest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import com.jayway.jsonpath.JsonPath;
+
+/**
+ * General ModelExecutor interface.
+ */
+public interface ModelExecutor {
+
+    default <T> ActionRequest getRemoteModelInferenceResult(Map<String, String> parameters, String modelId) {
+
+        RemoteInferenceInputDataSet inputDataSet = RemoteInferenceInputDataSet.builder().parameters(parameters).build();
+        if (inputDataSet.getParameters() == null) {
+            throw new IllegalArgumentException("wrong input. The model input cannot be empty.");
+        }
+        MLInput mlInput = MLInput.builder().algorithm(FunctionName.REMOTE).inputDataset(inputDataSet).build();
+
+        ActionRequest request = new MLPredictionTaskRequest(modelId, mlInput, null);
+
+        return request;
+
+    }
+
+    default String getModelInputFieldValue(Object originalFieldValue) {
+        Gson gson = new Gson();
```

Comment:
Let's avoid creating Gson object anywhere, previously we agreed that we should always use the gson object in https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java#L43

Reply:
removed